apps/server/src/shared/repo/courseexternaltool/course-external-tool.repo.ts

```diff
@@ -1,28 +1,60 @@
-import { BaseRepo } from '@shared/repo';
-import { EntityName } from '@mikro-orm/core';
+import { BaseDORepo, EntityProperties } from '@shared/repo';
+import { EntityName, Reference } from '@mikro-orm/core';
 import { Injectable } from '@nestjs/common/decorators/core/injectable.decorator';
-import { ObjectId } from '@mikro-orm/mongodb';
-import { EntityId } from '@shared/domain/types/entity-id';
-import { CourseExternalTool } from '@shared/domain';
+import { Course, CourseExternalTool, ICourseExternalToolProperties, SchoolExternalTool } from '@shared/domain';
+import { EntityManager } from '@mikro-orm/mongodb';
+import { CourseExternalToolDO } from '../../domain/domainobject/external-tool/course-external-tool.do';
+import { ExternalToolRepoMapper } from '../externaltool/external-tool.repo.mapper';
+import { Logger } from '../../../core/logger';
 
 @Injectable()
-export class CourseExternalToolRepo extends BaseRepo<CourseExternalTool> {
+export class CourseExternalToolRepo extends BaseDORepo<
+	CourseExternalToolDO,
+	CourseExternalTool,
+	ICourseExternalToolProperties
+> {
+	constructor(
+		private readonly externalToolRepoMapper: ExternalToolRepoMapper,
+		protected readonly _em: EntityManager,
+		protected readonly logger: Logger
+	) {
+		super(_em, logger);
+	}
+
 	get entityName(): EntityName<CourseExternalTool> {
 		return CourseExternalTool;
 	}
 
-	async findBySchoolToolIdAndCourseId(schoolToolId: EntityId, courseId: EntityId): Promise<CourseExternalTool | null> {
-		return this._em.findOne(CourseExternalTool, {
-			schoolTool: new ObjectId(schoolToolId),
-			course: new ObjectId(courseId),
+	entityFactory(props: ICourseExternalToolProperties): CourseExternalTool {
+		return new CourseExternalTool(props);
+	}
+
+	async deleteBySchoolExternalToolIds(schoolExternalToolIds: string[]): Promise<number> {
```

Comment:
This should be replaced by `findBySchoolExternalToolIds()`. The deletion can then be executed using the general `delete(do)` method. (See comment in `BaseDO`)